components/eamxx/cime_config/eamxx_buildnml_impl.py

```diff
@@ -418,33 +417,28 @@ def check_value(elem, value):
             if op=="ne":
                 rhs = refine_type(tokens[1],force_type=vtype)
                 expect (v!=rhs,
-                    "Value '{}' for entry '{}' violates constraint '{} != {}'"
-                    .format(v,elem.tag,v,rhs))
+                        f"Value '{v}' for entry '{elem.tag}' violates constraint '{v} != {rhs}'")
             elif op=="le":
                 rhs = refine_type(tokens[1],force_type=vtype)
                 expect (v<=rhs,
-                    "Value '{}' for entry '{}' violates constraint '{} <= {}'"
-                    .format(v,elem.tag,v,rhs))
+                        f"Value '{v}' for entry '{elem.tag}' violates constraint '{v} <= {rhs}'")
             elif op=="lt":
                 rhs = refine_type(tokens[1],force_type=vtype)
                 expect (v<rhs,
-                    "Value '{}' for entry '{}' violates constraint '{} < {}'"
-                    .format(v,elem.tag,v,rhs))
+                        f"Value '{v}' for entry '{elem.tag}' violates constraint '{v} < {rhs}'")
             elif op=="ge":
                 rhs = refine_type(tokens[1],force_type=vtype)
                 expect (v>=rhs,
-                    "Value '{}' for entry '{}' violates constraint '{} >= {}'"
-                    .format(v,elem.tag,v,rhs))
+                        f"Value '{v}' for entry '{elem.tag}' violates constraint '{v} >= {rhs}'")
             elif op=="gt":
                 rhs = refine_type(tokens[1],force_type=vtype)
                 expect (v>rhs,
-                    "Value '{}' for entry '{}' violates constraint '{} > {}'"
-                    .format(v,elem.tag,v,rhs))
+                        f"Value '{v}' for entry '{elem.tag}' violates constraint '{v} > {rhs}'")
             elif op=="mod":
                 expect (vtype=="integer",
-                    "Cannot evaluate constraint '{} mod {}' for entry '{}'\n"
-                    .format(lhs,tokens[1],elem.tag) +
-                    "Modulo constraint only makes sense for integer parameters.")
+                        "Cannot evaluate constraint '{} mod {}' for entry '{}'\n"
```

Comment:
Not a big deal, but since you went with f strings everywhere else, we could use them here too, and drop `format()`...

Reply:
We have almost 400 instances of `.format` use in our scripts... I bet nearly all of those would be improved by using f string instead. Maybe there's a script somewhere that can do this automatically?

Reply:
Yeah, as I said, no big deal. I was just seeing that you changed many other strings in this fcn, so I thought you might have wanted this changed too. No need to though.